src/main/java/seedu/expense/ui/MainWindow.java

```diff
@@ -195,6 +195,12 @@ private CommandResult executeCommand(String commandText) throws CommandException
                 handleExit();
             }
 
+            if (!commandResult.isShowBudgetDisplay() || (commandResult.isShowHelp() && !budgetDisplay.isVisible())) {
```

Comment:
Don't really understand the need for the second part: 
` (commandResult.isShowHelp() && !budgetDisplay.isVisible())`